main.js

```diff
@@ -447,4 +368,320 @@ function removeComparison(event) {
   }
 }
 
+var writeLeaderBoard = function updateLeaderBoard(event) {
+  showElement($leaderboard)
+  removeElement($playerStats)
+  var leaderBoardList = leaderBoardArray
+  var leaderBoardDOM = document.querySelector('#leader-board')
+  var createDiv = document.createElement('div')
+  createDiv.id = 'leader-board-rows'
+  var leaderBoardDiv = document.querySelector('#leader-board-rows')
+  if (document.querySelector('.buttons') === null) {
+    leaderBoardDOM.appendChild(leaderBoardStatButtons())
+    leaderBoardDOM.appendChild(leaderBoardMMRButtons())
+    var $buttons = document.querySelectorAll('.buttons')[0]
+    var $buttons2 = document.querySelectorAll('.buttons')[1]
+    $buttons.addEventListener('click', writeLeaderBoard)
+    $buttons2.addEventListener('click', writeLeaderBoard)
+    leaderBoardDOM.appendChild(createDiv)
+  }
+  if (event.target.textContent !== null && (event.target.className === 'statButton' || event.target.className === 'mmrButton')) {
+    clearLeaderBoard(leaderBoardDiv)
+    leaderBoardDiv.appendChild(leaderBoardHeader(event.target.textContent))
+    if (event.target.textContent === 'Solo Duel') {
```

Comment:
There is a lot of repetition here.
You could try a `switch` statement or a lookup table instead of lots of `else if`s.
```js
var leaderBoardLookup = {
  'solo-duel': 0,
  doubles: 1,
  'solo-standard': 2,
  standard: 3,
  wins: 4,
  goals: 5,
  mvps: 6,
  saves: 7,
  shots: 8,
  assists: 9
}
var board = leaderBoardList[event.target.getAttribute('data-leader-board')]
for (var i = 0; i < 10; i++) {
  leaderBoardDiv.appendChild(leaderBoardRow(board[i], i))
}
```